api/src/main/java/org/pmiops/workbench/api/WorkspacesController.java

```diff
@@ -244,23 +245,39 @@ private static String generateRandomChars(String candidateChars, int length) {
     return sb.toString();
   }
 
-  private void setCdrVersionId(Workspace workspace,
-      org.pmiops.workbench.db.model.Workspace dbWorkspace) {
-    if (workspace.getCdrVersionId() != null) {
+  private void setCdrVersionId(org.pmiops.workbench.db.model.Workspace dbWorkspace, String cdrVersionId) {
+    if (cdrVersionId != null) {
       try {
-        CdrVersion cdrVersion = cdrVersionDao.findOne(Long.parseLong(workspace.getCdrVersionId()));
+        CdrVersion cdrVersion = cdrVersionDao.findOne(Long.parseLong(cdrVersionId));
         if (cdrVersion == null) {
           throw new BadRequestException(
-              String.format("CDR version with ID %s not found", workspace.getCdrVersionId()));
+              String.format("CDR version with ID %s not found", cdrVersionId));
         }
         dbWorkspace.setCdrVersion(cdrVersion);
       } catch (NumberFormatException e) {
         throw new BadRequestException(String.format(
-            "Invalid cdr version ID: %s", workspace.getCdrVersionId()));
+            "Invalid cdr version ID: %s", cdrVersionId));
       }
     }
   }
 
+  /**
+   * Sets user-editable research purpose detail fields.
```

Comment:
It is an important distinction that this is setting only user-editable details and I judged it to be unwieldy to encode this into the method name, but feel free to suggest something. CreateWorkspace currently has a bug on account of a lack of making this distinction (sending a follow-up PR to fix) and would allow a client to just set their workspace as approved upon creation.

Reply:
Nope, this is great. I just wanted to push on it to make sure.